Bottom: 69fc9430bd65433269275f6f2143c9e9e548b4f3
Top:    cf5d3a146245482c3c0d60d661861483422ac9c6
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-25 19:44:55 +0100

draw.c: Split "Gathering" from real drawing routines (Element names)


---

diff --git a/src/draw.c b/src/draw.c
index 75a6c8e..41561f8 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -194,6 +194,23 @@ Redraw (void)
   gui->invalidate_all ();
 }
 
+static void
+draw_element_name (ElementType *element)
+{
+  if ((TEST_FLAG (HIDENAMESFLAG, PCB) && gui->gui) ||
+      TEST_FLAG (HIDENAMEFLAG, element))
+    return;
+  if (doing_pinout || doing_assy)
+    gui->set_color (Output.fgGC, PCB->ElementColor);
+  else if (TEST_FLAG (SELECTEDFLAG, &ELEMENT_TEXT (PCB, element)))
+    gui->set_color (Output.fgGC, PCB->ElementSelectedColor);
+  else if (FRONT (element))
+    gui->set_color (Output.fgGC, PCB->ElementColor);
+  else
+    gui->set_color (Output.fgGC, PCB->InvisibleObjectsColor);
+  DrawTextLowLevel (&ELEMENT_TEXT (PCB, element), PCB->minSlk);
+}
+
 static int
 name_callback (const BoxType * b, void *cl)
 {
@@ -205,7 +222,7 @@ name_callback (const BoxType * b, void *cl)
     return 0;
 
   if (ON_SIDE (element, *side))
-    DrawElementName (element);
+    draw_element_name (element);
   return 0;
 }
 
@@ -318,7 +335,7 @@ element_callback (const BoxType * b, void *cl)
   int *side = cl;
 
   if (ON_SIDE (element, *side))
-    draw_element_package (element);
+    DrawElementPackage (element);
   return 1;
 }
 
@@ -1512,19 +1529,11 @@ DrawElement (ElementTypePtr Element)
 void
 DrawElementName (ElementTypePtr Element)
 {
-  if (gui->gui && TEST_FLAG (HIDENAMESFLAG, PCB))
-    return;
+  assert (Gathering);
+
   if (TEST_FLAG (HIDENAMEFLAG, Element))
     return;
-  if (doing_pinout || doing_assy)
-    gui->set_color (Output.fgGC, PCB->ElementColor);
-  else if (TEST_FLAG (SELECTEDFLAG, &ELEMENT_TEXT (PCB, Element)))
-    gui->set_color (Output.fgGC, PCB->ElementSelectedColor);
-  else if (FRONT (Element))
-    gui->set_color (Output.fgGC, PCB->ElementColor);
-  else
-    gui->set_color (Output.fgGC, PCB->InvisibleObjectsColor);
-  DrawTextLowLevel (&ELEMENT_TEXT (PCB, Element), PCB->minSlk);
+  DrawTextLowLevel (&ELEMENT_TEXT (PCB, Element), 0);
 }
 
 /* ---------------------------------------------------------------------------
@@ -1741,6 +1750,8 @@ EraseElementPinsAndPads (ElementTypePtr Element)
 void
 EraseElementName (ElementTypePtr Element)
 {
+  assert (Gathering);
+
   if (TEST_FLAG (HIDENAMEFLAG, Element))
     return;
   DrawTextLowLevel (&ELEMENT_TEXT (PCB, Element), PCB->minSlk);
@@ -1839,7 +1850,7 @@ static void
 draw_element (ElementTypePtr element)
 {
   draw_element_package (element);
-  DrawElementName (element);
+  draw_element_name (element);
   DrawElementPinsAndPads (element);
 }
